openedx/features/partners/views.py

```diff
@@ -71,3 +102,250 @@ def reset_password_view(request):
+
+
+class PartnerRegistrationView(RegistrationViewCustom):
+    """
+    This class handles registration flow for give2asia users. It inherit some basic functionality from original (normal)
+    registration flow
+    """
+
+    def get(self, request, partner):
+        # Overriding get method to suppress corresponding method of parent class
+        raise Http404
+
+    def post(self, request, partner):
+        registration_data = request.POST.copy()
+
+        # validate data provided by end user
+        # PhilU implementation to import custom form if it exists
+        # Note that the override form must be named `AccountCreationFormCustom`
+        form = import_form_using_slug(partner.slug)
+        account_creation_form = form.AccountCreationFormCustom(data=registration_data, tos_required=True) if form \
+            else PartnerAccountCreationForm(data=registration_data, tos_required=True)
+        if not account_creation_form.is_valid():
+            return JsonResponse({"Error": dict(account_creation_form.errors.items())}, status=400)
+
+        account_creation_form.clean_registration_data(registration_data)
+        try:
+            # Create or update models for User, UserProfile,
+            # UserExtendedProfile, Organization, PartnerUser and EmailPreference
+            user = create_account_with_params_custom(request, registration_data, partner)
+            self.save_user_utm_info(user)
+        except Exception as err:
+            error_message = {"Error": {"reason": "User registration failed due to {}".format(repr(err))},
+                             "Partner": partner.label}
+            log.exception(error_message)
+            return JsonResponse(error_message, status=400)
+
+        response = JsonResponse({"success": True})
+        set_logged_in_cookies(request, response, user)
+        return response
+
+
+def create_account_with_params_custom(request, params, partner):
+    # Copy params so we can modify it; we can't just do dict(params) because if
+    # params is request.POST, that results in a dict containing lists of values
+    params = dict(params.items())
+
+    first_name, last_name = params['name']
+    # Perform operations within a transaction that are critical to account creation
+    with transaction.atomic():
+        try:
+            # Create user and activate it
+            user = User(
+                username=params['username'],
+                email=params["email"],
+                first_name=first_name,
+                last_name=last_name,
+                is_active=True,
+            )
+            user.set_password(params["password"])
+            registration = Registration()
+
+            user.save()
+            registration.register(user)
+        except Exception as err:  # pylint: disable=broad-except
+            log.exception("User creation failed for user {username}".format(username=params['username']), repr(err))
+            raise
```

Comment:
@mabdullahz it is good to raise your custom exception type to use specialized class.

Reply:
Totally agreed, but since this code was moved here I am not sure of the impact it'd have on the rest of the code. I will wait for the rest of the reviewers to go thru this and give suggestions regarding this if they have any.

Reply:
On second thought, I have implemented this. It looks much cleaner now, and the code makes more sense. Thank you for his suggestion!